File: viewer/ViewerEvent.mjs
/**
 * @typedef {"onViewerOpenLeft"
 *  | "onViewerOpenRight"
 * } ViewerEvent
 */

/** @type {ViewerEvent} */
const onViewerOpenLeft = "onViewerOpenLeft";

/** @type {ViewerEvent} */
const onViewerOpenRight = "onViewerOpenRight";

const ViewerEvent = Object.freeze({
  onViewerOpenLeft,
  onViewerOpenRight,
});

export default ViewerEvent;

File: viewer/src/main/scala/farjs/viewer/ViewerEvent.scala
package farjs.viewer

import scala.scalajs.js
import scala.scalajs.js.annotation.JSImport

@js.native
@JSImport("../viewer/ViewerEvent.mjs", JSImport.Default)
object ViewerEvent extends js.Object {

  val onViewerOpenLeft: String = js.native
  val onViewerOpenRight: String = js.native
}
